plugins/store/src/store.rs

```diff
@@ -174,42 +173,119 @@ impl<R: Runtime> StoreBuilder<R> {
         self
     }
 
-    /// Builds the [`Store`].
+    /// Disable auto save on modified with a debounce duration
+    pub fn disable_auto_save(mut self) -> Self {
+        self.auto_save = None;
+        self
+    }
+
+    /// Skip loading the store on build
+    pub fn skip_initial_load(mut self) -> Self {
+        self.load_on_build = false;
+        self
+    }
+
+    pub(crate) fn build_inner(mut self) -> crate::Result<(Arc<Store<R>>, ResourceId)> {
+        let state = self.app.state::<StoreState>();
+        let mut stores = state.stores.lock().unwrap();
+
+        if stores.contains_key(&self.path) {
+            return Err(crate::Error::AlreadyExists(self.path));
+        }
+
+        let mut store_inner = StoreInner::new(
+            self.app.clone(),
+            self.path.clone(),
+            self.defaults.take(),
+            self.serialize_fn,
+            self.deserialize_fn,
+        );
+        if self.load_on_build {
+            let _ = store_inner.load();
+        }
+
+        let store = Store {
+            auto_save: self.auto_save,
+            auto_save_debounce_sender: Arc::new(Mutex::new(None)),
+            store: Arc::new(Mutex::new(store_inner)),
+        };
+
+        let store = Arc::new(store);
+        let rid = self.app.resources_table().add_arc(store.clone());
+        stores.insert(self.path, rid);
+
+        Ok((store, rid))
+    }
+
+    pub(crate) fn build_or_existing_inner(mut self) -> (Arc<Store<R>>, ResourceId) {
+        let state = self.app.state::<StoreState>();
+        let mut stores = state.stores.lock().unwrap();
+
+        if let Some(rid) = stores.get(&self.path) {
+            (self.app.resources_table().get(*rid).unwrap(), *rid)
+        } else {
+            let mut store_inner = StoreInner::new(
+                self.app.clone(),
+                self.path.clone(),
+                self.defaults.take(),
+                self.serialize_fn,
+                self.deserialize_fn,
+            );
+            if self.load_on_build {
+                let _ = store_inner.load();
+            }
+
+            let store = Store {
+                auto_save: self.auto_save,
+                auto_save_debounce_sender: Arc::new(Mutex::new(None)),
+                store: Arc::new(Mutex::new(store_inner)),
+            };
+
+            let store = Arc::new(store);
+            let rid = self.app.resources_table().add_arc(store.clone());
+            stores.insert(self.path, rid);
+            (store, rid)
+        }
```

Comment:
this should call `self.build_inner()` and `build_or_existing` should return a result as well

Reply:
how about?
```rs
     {
        let state = self.app.state::<StoreState>();
        let mut stores = state.stores.lock().unwrap();

        if let Some(rid) = stores.get(&self.path) {
            return (self.app.resources_table().get(*rid).unwrap(), *rid);
        }
     }    
     
     self.build_inner()  
```

Reply:
If we change it to return a result then this will work, before this, we can't guarantee the thread safety